grl/policy_eval.py

```diff
@@ -50,29 +55,39 @@ def solve_mdp(self, mdp):
             a_t = a_t.at[s].set(-1)
             b_t = 0
             T_pi = np.tensordot(self.pi_ground[s], mdp.T, axes=1)
-            R_pi = np.tensordot(self.pi_ground[s], mdp.R, axes=1)
+            R_pi = np.tensordot(self.pi_ground[s], mdp.R * mdp.T, axes=1)
             for next_s in range(mdp.n_states):
                 t = T_pi[s,next_s]
                 r = R_pi[s,next_s]
                 # a_t[next_s] += t * mdp.gamma # add V_pi(s') to right side
                 a_t = a_t.at[next_s].set(a_t[next_s] + t * mdp.gamma)
-                b_t -= t * r # subtract constants to left side
+                b_t -= r # subtract constants to left side
```

Comment:
Before, `t` and `r` both came from quantities that had action probabilities multiplied through them. So when we do `t * r`, we are double counting them. So this works I think, but I'm trying to think of a more mathematically compatible way to format the code.

Reply:
Oh, I see!

Reply:
@dsctt said:
> Thinking about it more and following up on a comment, isn’t it necessarily the case that `R_pi` depends on `T` since you’d need to know what happens in the MDP after you choose an action before you can determine what reward you get? Maybe this is due to it being `R(s,a,s')` so it depends on `T` in the general case

I was skeptical, so we investigated...

![IMG_6358](https://user-images.githubusercontent.com/9636544/175097544-e0da3b83-d95f-4dff-8abd-37b7f8f56917.jpg)

And yes, `R_pi` definitely depends on `T`.